### PR TITLE
52230 - create catalog file part1

### PR DESCRIPTION
**Pull request by abhijeet14992**
Code changes and Unit tests
Functional Test

### CODE REVIEW

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 1]
```diff
@@ -133,9 +133,20 @@ public async Task<string> CreateLargeExchangeSet(SalesCatalogueServiceResponseQu
...
             });
+
             await Task.WhenAll(ParallelCreateFolderTasks);
             ParallelCreateFolderTasks.Clear();
 
+            List<Task> ParallelCreateFolderTaskForCatlogFile = new List<Task> { };
+            var dvdNumbersForCatlogFiles = Enumerable.Range(1, 2);
+            Parallel.ForEach(dvdNumbersForCatlogFiles, dvdNumbersForCatlogFile =>
+            {
+                ParallelCreateFolderTaskForCatlogFile.Add(LargeMediaExchangesetCatalogFile(message.BatchId, Path.Combine(homeDirectoryPath, currentUtcDate, message.BatchId, String.Format(largeExchangeSetFolderName, dvdNumbersForCatlogFile.ToString())), message.CorrelationId));
```
[the remark this answers is not in the document]
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs]: Resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 2]
```diff
@@ -409,5 +420,26 @@ private async Task<bool> CreateLargeMediaSerialEncFile(string batchId, string ex
                       },
                   batchId, correlationId);
         }
+
+        public async Task LargeMediaExchangesetCatalogFile(string batchId, string exchangeSetPath, string correlationId)
```
[the remark this answers is not in the document]
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs]: Resolved

**Review by harshal11869 — APPROVED** [review]
LGTM

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs [hunk 3]
```diff
@@ -348,5 +349,39 @@ public async Task WhenValidCreateLargeMediaSerialEncFileRequest_ThenReturnTrueRe
             Assert.AreEqual(true, response);
         }
         #endregion
+
+        #region CreateLargeExchangeSetCatalogFile
+        [Test]
+        public async Task WhenValidCreateLargeExchangeSetCatalogFileRequest_ThenReturnTrueReponse()
+        {
+            var b1 = A.Fake<IDirectoryInfo>();
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs:357]: What's the point of the b1 variable? It's just reassigned to directoryInfos and then that variable is used. Why not just set directoryInfos  = A.Fake<IDirectoryInfo>();

**Review by ajasonca — CHANGES_REQUESTED** [review]
Just an unused additional variable to remove.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs (hunk 3 above)
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs]: Resolved

**Review by ajasonca — APPROVED** [review]
Looks good

**Pull request merged**